### DISCUSSION TITLE
Is udp/DTLS supported in v9?

### DISCUSSION THREAD

**Discussion by kurtxiangli** [mitmproxy/mitmproxy#6057] [Q&A] [answered]
Hi,

I went through the release note of v9 and found the feature "Raw UDP and DTLS Support" is already introduced. And the "Available Options" on Doc also mentions the following two options are available, "rawudp" and "udp-hosts". However, I always encounter the following errors when trying to have them enabled. By checking the available options with '--help', I don't see these options out there. I did the google search as well as the discussions search, only found the discussion "Where is my local traffic? #5723" which seems resolved by WireGuard but not my case. Did I miss something? Thanks.

root@kurt-mint-21:~# ./mitmproxy -k   --mode transparent --showhost  --rawudp
usage: mitmproxy [options]
mitmproxy: error: unrecognized arguments: --rawudp

root@kurt-mint-21:~# ./mitmproxy -k   --mode transparent --showhost  --udp-hosts   0.0.0.0
usage: mitmproxy [options]
mitmproxy: error: unrecognized arguments: --udp-hosts 0.0.0.0

./mitmproxy --version
Mitmproxy: 9.0.1 binary
Python:    3.11.0
OpenSSL:   OpenSSL 3.0.7 1 Nov 2022
Platform:  Linux-5.15.0-56-generic-x86_64-with-glibc2.35
 lsb_release -a
No LSB modules are available.
Distributor ID:	Linuxmint
Description:	Linux Mint 21.1
Release:	21.1
Codename:	vera

**Comment by mhils** [comment, marked as answer]
> Is udp/DTLS supported in v9?

Yes. It depends on the proxy mode though, UDP is currently (mitmproxy 9) only supported in WireGuard and reverse proxy mode. IIRC we don't support UDP in transparent mode yet because that introduces significant complexity.

---------------

> However, I always encounter the following errors when trying to have them enabled.

From https://docs.mitmproxy.org/stable/concepts-options/#tools:

> For all tools, options can be set directly by name using the `--set` command-line option. 

You want to use `--set udp_hosts=example.com` instead of `--udp-hosts`.

**Reply by kurtxiangli** [reply]
Thanks. This may be closed.
